drivers/sensor/adxrs290/adxrs290.c

```diff
@@ -0,0 +1,416 @@
+			change = true;
+		}
+	} else {
+		/* current status off -> setting measurement mode */
+		if (!standByMode) {
+			data |= ADXRS290_POW_CTRL_STDBY_MASK;
+			change = true;
+		}
+	}
+	if (change) {
+		adxrs290_write_register(ADXRS290_POW_CTRL_REG, data);
+		k_sleep(K_MSEC(100));
+	}
+}
+
+void adxrs290_interrupt_mode_enable(uint8_t activate)
+{
+	uint8_t data;
+
+	data = adxrs290_read_register(ADXRS290_DATA_READY_REG);
+
+	data &= ~ADXRS290_DATA_READY_INT_MASK;
+	if (activate) {
+		data |=  0x01;
+	}
+	adxrs290_write_register(ADXRS290_DATA_READY_REG, data);
+}
+
+int adxrs290_get_low_pass_filter(void)
+{
+	uint8_t data;
+
+	data = adxrs290_read_register(ADXRS290_BANDPASS_FILTER);
+	return (data & ADXRS290_BPF_LPF_MASK);
+}
+
+void adxrs290_set_low_pass_filter(int lowFreqPole)
+{
+	uint8_t data;
+
+	data = adxrs290_read_register(ADXRS290_BANDPASS_FILTER);
+	data &= ~ADXRS290_BPF_LPF_MASK;
+	data |= (lowFreqPole & ADXRS290_BPF_LPF_MASK);
+	adxrs290_write_register(ADXRS290_BANDPASS_FILTER, data);
+}
+
+int adxrs290_get_high_pass_filter(void)
+{
+	uint8_t data;
+
+	data = adxrs290_read_register(ADXRS290_BANDPASS_FILTER);
+	return ((data & ADXRS290_BPF_HPF_MASK) >> ADXRS290_BPF_HPF_OFFSET);
+}
+
+void adxrs290_set_high_pass_filter(int highFreqPole)
+{
+	uint8_t data;
+
+	data = adxrs290_read_register(ADXRS290_BANDPASS_FILTER);
+	data &= ~ADXRS290_BPF_HPF_MASK;
+	data |= ((highFreqPole << ADXRS290_BPF_HPF_OFFSET) &
+			  ADXRS290_BPF_HPF_MASK);
+	adxrs290_write_register(ADXRS290_BANDPASS_FILTER, data);
+}
+
+void adxrs290_temp_sensor_enable(uint8_t enable)
+{
+	uint8_t data;
+
+	data = adxrs290_read_register(ADXRS290_POW_CTRL_REG);
+
+	data &= ~ADXRS290_POW_CTRL_TEMP_EN_MASK;
+	if (enable) {
+		data |=  ADXRS290_POW_CTRL_TEMP_EN_MASK;
+	}
+	adxrs290_write_register(ADXRS290_POW_CTRL_REG, data);
+}
+
+void adxrs290_read_xy(int16_t *x, int16_t *y)
+{
+
+	if (adxrs290_data.standby) {
+		return adxrs290_standby_read_xy(x, y);
+	}
+
+	adxrs290_read_n_registers(ADXRS290_GYR_X_L, readBuf, 4);
+	*x = readBuf[2] | (readBuf[3] << 8);
+	*y = readBuf[0] | (readBuf[1] << 8);
+}
+
+int16_t adxrs290_read_temperature(void)
+{
+	int16_t data;
+
+	if (adxrs290_data.standby) {
+		return adxrs290_standby_read_temperature();
+	}
+
+	adxrs290_read_n_registers(ADXRS290_TEMP_L, readBuf, 2);
+	data = readBuf[0] | ((0x0F & readBuf[1]) << 8);
+
+	return data;
+}
+
+static int adxrs290_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct adxrs290_data *data = dev->driver_data;
+
+	adxrs290_read_xy(&data->x, &data->y);
+	data->temp = adxrs290_read_temperature();
+
+	return 0;
+}
+
+static int adxrs290_channel_get(struct device *dev,
+					  enum sensor_channel chan,
+					  struct sensor_value *val)
+{
+	struct adxrs290_data *data = dev->driver_data;
+
+	switch (chan) {
+
+	case SENSOR_CHAN_GYRO_X:
+		/* Angular velocity around the X axis, in radians/s. */
+		val->val1 = (int32_t)data->x / 200;
+		val->val2 = ((int32_t)data->x % 200) * 1000000;
```

Comment:
Since you are dividing by 200, you should multiply by 1000000/200. Same for the next two.